numalogic/connectors/_config.py

```diff
@@ -52,6 +54,49 @@ def __post_init__(self):
             self.aggregations = {"count": doublesum("count")}
 
 
+@dataclass
+class RDSFetcherConf:
+    """
+    RDSFetcherConf class represents the configuration for fetching data from an RDS data source.
+
+    Args:
+        datasource (str): The name of the data source.
+        dimensions (list[str]): A list of dimension column names.
+        group_by (list[str]): A list of column names to group the data by.
+        pivot (Pivot): An instance of the Pivot class representing the pivot configuration.
+        hash_query_type (bool): A boolean indicating whether to use hash query type.
+        hash_column_name (Optional[str]): The name of the hash column. (default: None)
+        datetime_column_name (str): The name of the datetime column. (default: "eventdatetime")
+        metrics (list[str]): A list of metric column names.
+
+    Methods
+    -------
+        __post_init__(): Performs post-initialization validation checks.
+
+    Raises
+    ------
+        RDSFetcherConfValidationException: If the hash_query_type is enabled
+        but hash_column_name is not provided.
+    """
+
+    datasource: str
+    dimensions: list[str] = field(default_factory=list)
+    group_by: list[str] = field(default_factory=list)
+    pivot: Pivot = field(default_factory=lambda: Pivot())
+    hash_query_type: bool = field(default=True)
+    hash_column_name: Optional[str] = field(default_factory=str)
+    datetime_column_name: str = "eventdatetime"
+    # metric column names
+    metrics: list[str] = field(default_factory=list)
+
+    def __post_init__(self):
+        if self.hash_query_type:
+            if not self.hash_column_name or self.hash_column_name.strip() == "":
```

Comment:
```suggestion
            if not self.hash_column_name:
```
This should work right?